### ISSUE TITLE
resolve starts to fail after running for a while -  object has no attribute 'writeHeaders'

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry#1716]
## The Issue
Was running the daemon for a while and then it started failing on resolve requests randomly. 

A reboot fixed it. Not sure how it got into this state, but it was receiving blocks. 

```2018-12-14 10:44:44,557 WARNING  lbrynet.extras.daemon.auth.server:338: Cancelling dropped api request resolve
2018-12-14 10:44:44,560 ERROR    lbrynet.extras.daemon.auth.server:606: Failed to render API response: {'error': ''}
Traceback (most recent call last):
  File "lbrynet\extras\daemon\auth\server.py", line 604, in _callback_render
  File "lbrynet\extras\daemon\auth\server.py", line 303, in _render_message
  File "twisted\web\server.py", line 237, in write
  File "twisted\web\http.py", line 1122, in write
AttributeError: 'NoneType' object has no attribute 'writeHeaders'
2018-12-14 10:44:44,586 WARNING  lbrynet.extras.daemon.auth.server:325: error processing api request: 'NoneType' object has no attribute 'write'
traceback:   File "twisted\internet\defer.py", line 654, in _runCallbacks
    
  File "lbrynet\extras\daemon\auth\server.py", line 96, in trap
    
  File "twisted\python\failure.py", line 439, in trap
    
  File "twisted\python\failure.py", line 467, in raiseException
    
  File "twisted\internet\defer.py", line 654, in _runCallbacks
    
  File "lbrynet\extras\daemon\auth\server.py", line 607, in _callback_render
    
  File "lbrynet\extras\daemon\auth\server.py", line 333, in _render_error
    
  File "lbrynet\extras\daemon\auth\server.py", line 303, in _render_message
    
  File "twisted\web\server.py", line 237, in write
    
  File "twisted\web\http.py", line 1139, in write
    
builtins.AttributeError: 'NoneType' object has no attribute 'write'
2018-12-14 10:46:00,410 INFO     lbrynet.extras.daemon.Daemon:823: Get version info: {"processor": "Intel64 Family 6 Model 158 Stepping 9, GenuineIntel", "python_version": "3.7.1", "platform": "Windows-10-10.0.17134", "os_release": "10", "os_system": "Windows", "lbrynet_version": "0.30.1", "lbryschema_version": "0.0.16", "build": "release"}
2018-12-14 10:46:02,130 INFO     lbrynet.extras.daemon.Daemon:823: Get version info: {"processor": "Intel64 Family 6 Model 158 Stepping 9, GenuineIntel", "python_version": "3.7.1", "platform": "Windows-10-10.0.17134", "os_release": "10", "os_system": "Windows", "lbrynet_version": "0.30.1", "lbryschema_version": "0.0.16", "build": "release"}
2018-12-14 10:46:02,136 INFO     lbrynet.extras.daemon.Daemon:823: Get version info: {"processor": "Intel64 Family 6 Model 158 Stepping 9, GenuineIntel", "python_version": "3.7.1", "platform": "Windows-10-10.0.17134", "os_release": "10", "os_system": "Windows", "lbrynet_version": "0.30.1", "lbryschema_version": "0.0.16", "build": "release"}
2018-12-14 10:46:02,863 INFO     lbrynet.extras.daemon.DaemonControl:65: Starting lbrynet-daemon from command line
2018-12-14 10:46:02,910 INFO     lbrynet.extras.daemon.auth.factory:48: Using non-authenticated API
2018-12-14 10:46:02,910 INFO     lbrynet.extras.daemon.auth.server:222: Not using SSL
2018-12-14 10:46:02,911 ERROR    lbrynet.extras.daemon.auth.server:231: lbrynet API failed to bind TCP localhost:5279 for listening. Daemon is already running or this port is already in use by another application.
2018-12-14 10:46:08,936 WARNING  lbrynet.extras.daemon.auth.server:338: Cancelling dropped api request account_balance
2018-12-14 10:46:08,943 WARNING  lbrynet.extras.daemon.auth.server:325: error processing api request: CancelledError
traceback:   File "twisted\internet\defer.py", line 654, in _runCallbacks
    
  File "lbrynet\extras\daemon\auth\server.py", line 96, in trap
    
  File "twisted\python\failure.py", line 439, in trap
    
  File "twisted\python\failure.py", line 467, in raiseException
    
  File "twisted\internet\defer.py", line 824, in adapt
    
concurrent.futures._base.CancelledError: 
2018-12-14 10:46:11,682 INFO     torba.client.baseledger:158: lbc_mainnet: added BlockHeightEvent(height=485889, change=1) header blocks, final height 485889
```

## System Configuration

 

- LBRY Daemon version:
- LBRY App version:
- LBRY Installation ID:
- Operating system:

### Acceptance Criteria
1.
2.
3.

### Definition of Done
- [ ]  Tested against acceptance criteria
- [ ] Tested against the assumptions of user story
- [ ] The project builds without errors
- [ ] Unit tests are written and passing
- [ ] Tests on devices/browsers listed in the issue have passed
- [ ] QA performed & issues resolved
- [ ] Refactoring completed
- [ ] Any configuration or build changes documented
- [ ] Documentation updated
- [ ] Peer Code Review performed

**Comment by tzarebczan** [comment]
I think this may be related to the bug fixed by https://github.com/lbryio/lbry/pull/1718 - we'll see if it goes away once we get a new daemon out. I see others with the same issue on loggly.

**Comment by eukreign** [comment]
@jackrobison can this scenario occur in the new blob exchange implementation?

**Comment by tzarebczan** [comment]
This still seems to be happening on 0.30.4 (according to loggly - but I'll dig in more later), so don't think it was related to #1718 . Not sure what issues it causes yet, haven't run into it myself.

**Comment by shyba** [comment]
@tzarebczan @eukreign we arent using Twisted anymore, can this be updated?
Failed resolves is an issue, but the error here is just a dropped/cancelled/timed-out API call. This is #1619 today, being addressed on #1952 
For this specific issue here, what could be investigated is if the wallet fails to reconnect or something like that, but the stack traces in here arent related, they are triggered when the caller cancels by its own timeout.
